src/main/java/org/opentripplanner/routing/algorithm/RoutingWorker.java

```diff
@@ -40,20 +43,39 @@
 
     private final RoutingRequest request;
     private final Router router;
-    private final FilterTransitWhenDirectModeIsEmpty emptyDirectModeHandler;
+    private FilterTransitWhenDirectModeIsEmpty emptyDirectModeHandler;
+    private final ZonedDateTime searchStartTime;
+    private SearchParams raptorSearchParamsUsed = null;
     private Itinerary firstRemovedItinerary = null;
-    private Instant filterOnLatestDepartureTime = null;
-    private int searchWindowUsedInSeconds = TransitRouter.NOT_SET;
+    private boolean reverseFilteringDirection = false;
 
-    public RoutingWorker(RoutingRequest request, Router router) {
+    public RoutingWorker(Router router, RoutingRequest request, ZoneId zoneId) {
         this.request = request;
         this.router = router;
-        this.emptyDirectModeHandler = new FilterTransitWhenDirectModeIsEmpty(request.modes);
+        this.searchStartTime = DateMapper.asStartOfService(request.getDateTimeCurrentPage(), zoneId);
     }
 
     public RoutingResponse route() {
+        LOG.debug("RoutingWorker input");
+        LOG.debug("PageCursor        : " + request.pageCursor);
+        LOG.debug("Request sw        : " + request.searchWindow);
+        LOG.debug("Request org. time : " + request.getDateTimeOriginalSearch());
+        LOG.debug("Request page time : " + request.getDateTimeCurrentPage());
```

Comment:
In general we remove debug code in the main branch, if you would like to keep it at least hide it so it does not obstruct the main flow of the "highest abstraction level method". You can do so like this:
```
if(LOG.isDebug()) { debugRouterWorkerInput(request); }
```
and create a static debug method close to the bottom of the file.